src/main/java/com/gengo/client/GengoClient.java

```diff
@@ -352,9 +363,143 @@ public JSONObject getTranslationJobs() throws GengoException
+     * @param timestampAfter Unix epoch seconds from which to filter submitted jobs
+     * @param count limit count of jobs(default 10, maximum 200)
+     * @throws GengoException
+     */
+    public JSONObject getTranslationJobs(List<Integer> ids, JobStatus status, Integer timestampAfter, Integer count) throws GengoException
+    {
+        String url = getBaseUrl() + "translate/jobs/";
+        JSONObject data = new JSONObject();
+        if (ids != null && ids.size() > 0) {
+            url += join(ids, ",");
+        }
+        if (status != null) {
+            try {
+                data.put("status", status.getStatusString());
+            } catch (JSONException e) {
+                throw new GengoException(e.getMessage(), e);
+            }
+        }
+        if (timestampAfter != null && timestampAfter > 0) {
+            try {
+                data.put("timestamp_after", timestampAfter);
+            } catch (JSONException e) {
+                throw new GengoException(e.getMessage(), e);
+            }
+        }
+        if (count != null && count > 0) {
+            try {
+                data.put("count", count);
+            } catch (JSONException e) {
+                throw new GengoException(e.getMessage(), e);
+            }
+        }
+        return call(url, HttpMethod.GET, data);
+    }
+
+    /**
+     * Set revision comment for each jobs.
+     * @param jobs each Entry consists of job ID and comment text
+     * @throws GengoException
+     */
+    public JSONObject reviseTranslationJobs(HashMap<Integer, String> jobs) throws GengoException
+    {
+        try
+        {
+            String url = getBaseUrl() + "translate/jobs";
+            JSONObject data = new JSONObject();
+            data.put("action", "revise");
+            // [begin] Generate 'job_ids' parameter.
+            JSONArray job_ids = new JSONArray();
+            for (Entry<Integer, String> iJob : jobs.entrySet()) {
+                HashMap<String, Object> job = new HashMap<String, Object>();
+                job.put("job_id", iJob.getKey());
+                job.put("comment", iJob.getValue());
+                job_ids.put(job);
+            }
+            // [end] Generate 'job_ids' parameter.
+            data.put("job_ids", job_ids);
+            return call(url, HttpMethod.PUT, data);
+        } catch (JSONException e)
+        {
+            throw new GengoException(e.getMessage(), e);
+        }
+    }
+
+    /**
+     * Approve jobs with collateral attributes.
+     * @param jobs list of Approval instance
+     * @throws GengoException
+     */
+    public JSONObject approveTranslationJobs(List<Approval> jobs) throws GengoException
+    {
+        try
+        {
+            String url = getBaseUrl() + "translate/jobs";
+            JSONObject data = new JSONObject();
+            data.put("action", "approve");
+            JSONArray iJobs = new JSONArray();
+            for (int i = 0; i < jobs.size(); i++) {
```

Comment:
enhanced for is better.

But you can work on it in another PR because it is unlikely that the change breaks something and we have to merge this PR earlier for a certain reason.